File: ftp/ruby/delete/ftp_delete.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	ftp_delete.rb
#
#					Jan/23/2012
# --------------------------------------------------------------------------
require 'rubygems'
require 'net/ftp'
require 'json'
#
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
load '/var/www/data_base/common/ruby_common/net_manipulate.rb'
#
# --------------------------------------------------------------------------
puts "*** 開始 ***"
key_in = ARGV[0]
#
puts key_in
#
host='cddn007'
user='scott'
pass='tiger'
target_file='city/iwate.json'
#
json_str = ftp_get_proc(host,user,pass,target_file)
#
dict_aa=JSON.parse(json_str)
#
if (dict_aa.key?(key_in))
	dict_aa = dict_delete_proc(dict_aa,key_in)
	dict_display_proc(dict_aa)
	str_json_out = JSON.pretty_generate(dict_aa)
	ftp_put_proc(host,user,pass,target_file,str_json_out)
end
#
#
puts	"*** 終了 ***"
#
# --------------------------------------------------------------------------

File: ftp/jython/delete/ftp_delete.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	ftp_delete.py
#
#					Feb/25/2013
#
# -------------------------------------------------------------
from ftplib import FTP
import	sys
import	os
import	string
import	json
#
sys.path.append ('/var/www/data_base/common/jython_common')
from jython_file_io import file_to_str_proc
from jython_ftp_manipulate import ftp_get_proc
from jython_ftp_manipulate import ftp_put_proc
#
sys.path.append ('/var/www/data_base/common/python_common')
from text_manipulate import dict_delete_proc
#
# -------------------------------------------------------------
print	("*** 開始 ***")
id_in = sys.argv[1]
print ("%s" % id_in)
#
server='host_dbase'
user='scott'
passwd='tiger'
path='city/iwate.json'
#
json_str = ftp_get_proc (server,user,passwd,path)
#
dict_aa = json.loads (json_str)
#
dict_aa = dict_delete_proc (dict_aa,id_in)
#
json_str = json.dumps (dict_aa)
#
ftp_put_proc (server,user,passwd,path,json_str)
#
print	("*** 終了 ***")
# -------------------------------------------------------------
